src/yunohost/tools.py

```diff
@@ -530,6 +530,10 @@ def tools_upgrade(operation_logger, auth, ignore_apps=False, ignore_packages=Fal
     is_api = True if msettings.get('interface') == 'api' else False
 
     if not ignore_packages:
+
+        apt.apt_pkg.config.init()
```

Comment:
```suggestion
        apt.apt_pkg.init()
```

You need to init apt_pkg, not the config (while this will init the "main config" and the "system" ...)

Reply:
Indeed :+1: